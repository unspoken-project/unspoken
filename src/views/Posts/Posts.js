import React, { useEffect, useState } from 'react';
import Preview from '../../components/Preview/Preview';
import './Posts.css';
import { getPosts } from '../../services/posts';

export default function Posts() {
  const [posts, setPosts] = useState([]);
  const [loading2, setLoading2] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const resp = await getPosts();
      setPosts(resp);
      setLoading2(false);
    };
    fetchData();
  }, []);

  if (loading2) {
    return <h3>Loading...</h3>;
  }

  return (
    <>
      <div className="post-list">
        {posts.map((item) => (
          <Preview key={item.id} {...item} />
        ))}
      </div>
    </>
  );
}
