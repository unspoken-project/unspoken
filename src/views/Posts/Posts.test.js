import { MemoryRouter } from 'react-router-dom';
import Posts from '../Posts/Posts';
import { render, screen } from '@testing-library/react';

test.skip('should render list of posts', async () => {
  const { container } = render(
    <MemoryRouter>
      <Posts />
    </MemoryRouter>
  );
  await screen.findByText('Dear Julie...');
  expect(container).toMatchSnapshot();
});
