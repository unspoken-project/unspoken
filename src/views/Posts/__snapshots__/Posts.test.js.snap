// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list of posts 1`] = `
<div>
  <div
    class="post-list"
  >
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/67"
        >
          <div
            class="title"
          >
            Julie...what can I say
          </div>
        </a>
        <div
          class="content"
        >
          more like, what I can do. You've changed my life. I know you might not want all the credit, but I wouldn't have gotten this far without you. You were the key to the start of the rest of my life. And I look forward to seeing you throughout the future.
          <p>
            <a
              href="/posts/67"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/68"
        >
          <div
            class="title"
          >
            based julie
          </div>
        </a>
        <div
          class="content"
        >
          dang what an amazing role model!!!!!!!! life changing
          <p>
            <a
              href="/posts/68"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/69"
        >
          <div
            class="title"
          >
            here are some reasons why you rule 
          </div>
        </a>
        <div
          class="content"
        >
          to echo what michelle said, learning this new technical, abstract, challenging stuff from someone who isn't a man was a powerful experience for me, too. you made me believe i could learn it. you really believed in me when i didn't believe in myself. without your encouragement and vote of confidence i don't know if i would have made it this far. you're a fantastic teacher, not only do you have extensive knowledge, you're also able to relay it in a way that makes sense to those of us with no experience. that is something many smart people lack. your emotional intelligence mirrors your technical skill, and that is a rarity in the professional world. you're a GEM! thank you for ushering us into the next chapter of our lives with grace, patience, humor, and empathy! can't thank you enough. marty! give this rock star a RAISE!!!!!!!
          <p>
            <a
              href="/posts/69"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/70"
        >
          <div
            class="title"
          >
            Julie is a life changer. Thank you.
          </div>
        </a>
        <div
          class="content"
        >
          you have given us a priceless gift that will change our lives for generations to come. YOU are the reason I stayed with alchemy through the transition because you are such a rad teacher. Wishing you happiness and fortune for the rest of your and benny's days! &lt;3 
          <p>
            <a
              href="/posts/70"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/53"
        >
          <div
            class="title"
          >
            Dear Julie...
          </div>
        </a>
        <div
          class="content"
        >
          Thank you so much for everything you've taught me. Your genuine interest in our learning and passion for coding shine through every day, and kept me excited even when I was exhausted. I'll miss you so much!
          <p>
            <a
              href="/posts/53"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/58"
        >
          <div
            class="title"
          >
            Friend
          </div>
        </a>
        <div
          class="content"
        >
          Over 5 years?? Wild that people so far away can meet and learn about and know each other. One day I'll drive down to you again and we can HUG once more!!!
          <p>
            <a
              href="/posts/58"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/59"
        >
          <div
            class="title"
          >
            as a woman
          </div>
        </a>
        <div
          class="content"
        >
          I was worried about going into tech; the biases from others, the difficult questions they'd ask me, the internal conflicts. But being taught by Julie was the best introduction into this world I could have asked for. I saw how intelligent, powerful and graceful she is, and it inspired me to be as such. Im sure Julie has some stories too to have gotten to where she is now, and I hope she knows that she has become a part of mine.
          <p>
            <a
              href="/posts/59"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/60"
        >
          <div
            class="title"
          >
            what would have happened
          </div>
        </a>
        <div
          class="content"
        >
          What would have happened if I gave my 500%

Stepped on the gas pedal all the way

And allowed success and goodness in my life?
          <p>
            <a
              href="/posts/60"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/71"
        >
          <div
            class="title"
          >
            To Julie and TA's
          </div>
        </a>
        <div
          class="content"
        >
          Thank you ALL for providing such an amazing introduction to coding for me. It's been truly a one-of-a-kind experience and a time I will never forget. I've grown so much in these past 4 months technically / mentally speaking and it's all {props} to each one of you. Don't forget to visit Portland! p.s. this app is amazing.
          <p>
            <a
              href="/posts/71"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/72"
        >
          <div
            class="title"
          >
            Project Week!
          </div>
        </a>
        <div
          class="content"
        >
          We watched project presentations today and they were all so impressive, Alchemy is great!
          <p>
            <a
              href="/posts/72"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/61"
        >
          <div
            class="title"
          >
            Bye Julie
          </div>
        </a>
        <div
          class="content"
        >
          Dear Julie, 
Thank you so much for everything and for being an awesome teacher as well as a friend. Thank you for believing in me and pushing me to do my best. I will miss you so much and I am super sad you not going to be our teacher. xoxox Cancer babies 4ever
          <p>
            <a
              href="/posts/61"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/64"
        >
          <div
            class="title"
          >
            Dear Julie and the wonderful September Cohort
          </div>
        </a>
        <div
          class="content"
        >
          I am so glad we all get to have this experience together! Julie, thank you so much for being here to provide guidance for the very beginning of our coding journey! We couldn't have done it without you. And thank you September for being such a fun, wonderful, and bright group of people. I am so glad we get to learn together and help each other grow. &lt;3
          <p>
            <a
              href="/posts/64"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/66"
        >
          <div
            class="title"
          >
            we're the lucky ones
          </div>
        </a>
        <div
          class="content"
        >
          We're so lucky to have an instructor like Julie and a cohort like this. We've learned so much, including how to be supports for one another. Julie, KIT, and don't ever change. Come visit us!
          <p>
            <a
              href="/posts/66"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/74"
        >
          <div
            class="title"
          >
            You guys did great!
          </div>
        </a>
        <div
          class="content"
        >
          This app is great. You guys did such a good job on it. We're all very proud of you guys.
          <p>
            <a
              href="/posts/74"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/75"
        >
          <div
            class="title"
          >
            Julie you are a beast AND you are the best
          </div>
        </a>
        <div
          class="content"
        >
          Coding pro and instructor extraordinaire! Thanks for everything. I thought about quitting when the curriculum changed but I decided to stay because of you. 
          <p>
            <a
              href="/posts/75"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/76"
        >
          <div
            class="title"
          >
            Looks like people are saying nice things about Julie here so...
          </div>
        </a>
        <div
          class="content"
        >
          In the words of one of my favorite authors, "Be prepared to struggle, both intellectually and creatively. The truth is, I don't know of any human activity as demanding of one's imagination, intuition and ingenuity. But I do it anyway. I do it because I love it and because I can't help it. Once you've been to the jungle, you can never really leave. It haunts your waking dreams."

Thank you Julie for helping me in my struggle! You are an inspiration and a role model. It's nice to see you had such an impact on the September cohort, I know you did on mine as well.
          <p>
            <a
              href="/posts/76"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="title-link"
          href="/posts/78"
        >
          <div
            class="title"
          >
            Thank you...
          </div>
        </a>
        <div
          class="content"
        >
          I've loved teaching all of you. Thank you for these lovely words. Good luck in the next module -- I'll be rooting for all of you and can't wait to see what you build. 
          <p>
            <a
              href="/posts/78"
            >
              read more..
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
